open_instruct/ppo_vllm_thread_ray_gtrl.py

```diff
@@ -709,7 +712,7 @@ def from_pretrained(
             disable_dropout_in_model(self.reward_model)
             ds_config = get_eval_ds_config(
                 offload=False,
-                stage=args.deepspeed_stage,
+                stage=args.deepspeed_stage if args.deepspeed_stage == 3 else 0,
```

Comment:
This is also a little surprising. A little funny but must just be how it works. What does it fix?

Reply:
Yeah reference model / reward model does not have a stage 2.

Stage 2 is optimizer sharding. If we pass stage 2 like done for olmoe, then it errors out with stage 2 requiring an optimizer.

Reply:
I am gonna add a note there.